src/astro/databases/snowflake.py

```diff
@@ -75,3 +77,46 @@ def load_pandas_dataframe_to_table(
             chunk_size=chunk_size,
             quote_identifiers=False,
         )
+
+    def add_templates_to_context(self, parameters: Dict, context: Dict) -> Dict:
+        """
+        When running functions through the `aql.transform` and `aql.render` functions, we need to add
+        the parameters given to the SQL statement to the Airflow context dictionary. This is how we can
+        then use jinja to render those parameters into the SQL function when users use the {{}} syntax
+        (e.g. "SELECT * FROM {{input_table}}").
+
+        With this system we should handle Table objects differently from other variables. Since we will later
+        pass the parameter dictionary into SQLAlchemy, the safest (From a security standpoint) default is to use
+        a `:variable` syntax. This syntax will ensure that SQLAlchemy treats the value as an unsafe template. With
+        Table objects, however, we have to give a raw value or the query will not work. Because of this we recommend
+        looking into the documentation of your database and seeing what best practices exist (e.g. Identifier wrappers
+        in snowflake).
+
+        :param parameters: A Dict of SQL key-value parameters
+        :param context: Airflow Context dictionary
+        :return: Dictionary with values with Table type replaced with the table name
+        """
+        for k, v in parameters.items():
+            if isinstance(v, Table):
+                context[k] = "IDENTIFIER(:" + k + ")"
+            else:
+                context[k] = ":" + k
+        return context
+
+    def process_sql_parameters(self, parameters: Dict) -> Dict:
```

Comment:
Is there anything specific to the Database here - or could we keep this together with our base operator?